File: frontend-vue/src/stores/financialHealth.ts
import { defineStore } from "pinia";
import axios from "@/lib/api";

export interface FinancialHealthRatio {
  name: string;
  value: number;
  formatted_value: string;
  target: string;
  status: "Sehat" | "Waspada" | "Bahaya";
  description: string;
}

export interface FinancialHealthResponse {
  overall_score: number;
  overall_status: "Sehat" | "Waspada" | "Bahaya";
  ratios: FinancialHealthRatio[];
}

export const useFinancialHealthStore = defineStore("financialHealth", {
  state: () => ({
    data: null as FinancialHealthResponse | null,
    isLoading: false,
  }),
  actions: {
    async fetchFinancialHealth() {
      this.isLoading = true;
      try {
        const response = await axios.get("/api/financial-health");
        this.data = response.data.data;
      } catch (error) {
        console.error("Failed to fetch financial health:", error);
      } finally {
        this.isLoading = false;
      }
    },
  },
});

File: backend-go/internal/entity/financial_health.go
package entity

type FinancialHealthStatus string

const (
	StatusHealthy FinancialHealthStatus = "Sehat"
	StatusWarning FinancialHealthStatus = "Waspada"
	StatusDanger  FinancialHealthStatus = "Bahaya"
)

type FinancialHealthRatio struct {
	Name           string                `json:"name"`
	Value          float64               `json:"value"`
	FormattedValue string                `json:"formatted_value"`
	Target         string                `json:"target"`
	Status         FinancialHealthStatus `json:"status"`
	Description    string                `json:"description"`
}

type FinancialHealthResponse struct {
	OverallScore  float64                `json:"overall_score"`
	OverallStatus FinancialHealthStatus  `json:"overall_status"`
	Ratios        []FinancialHealthRatio `json:"ratios"`
}
